packages/sync-actions/src/product-actions.js

```diff
@@ -453,31 +459,38 @@ function _buildVariantImagesAction (diffedImages, oldVariant, newVariant) {
           actions.push({
             action: 'removeImage',
             variantId: oldVariant.id,
-            imageUrl: oldVariant.images[key].url,
+            imageUrl: oldVariant.images[imagesHashMap[key][0]].url,
           })
           actions.push({
             action: 'addExternalImage',
             variantId: oldVariant.id,
-            image: newVariant.images[key],
+            image: newVariant.images[imagesHashMap[key][1]],
           })
         } else if ({}.hasOwnProperty.call(image, 'label') &&
           (image.label.length === 1 || image.label.length === 2))
           actions.push({
             action: 'changeImageLabel',
             variantId: oldVariant.id,
-            imageUrl: oldVariant.images[key].url,
+            imageUrl: oldVariant.images[imagesHashMap[key][0]].url,
             label: diffpatcher.getDeltaValue(image.label),
           })
-    } else if (REGEX_UNDERSCORE_NUMBER.test(key)) {
-      const index = key.substring(1)
-
-      if (Array.isArray(image))
-        actions.push({
-          action: 'removeImage',
-          variantId: oldVariant.id,
-          imageUrl: oldVariant.images[index].url,
-        })
-    }
+    } else if (REGEX_UNDERSCORE_NUMBER.test(key))
+      if (Array.isArray(image) && image.length === 3)
+        if (Number(image[2]) === 3)
+          // image position changed
+          actions.push({
```

Comment:
Nice, this wasn't supported back in the days which is why the MC does create their own update actions in the image view.